gui/wxpython/datacatalog/tree.py

```diff
@@ -572,10 +572,29 @@ def OnDoubleClick(self, node):
         """
         if not isinstance(self._giface, StandaloneGrassInterface):
             self.DefineItems([node])
-            if self.selected_layer[0] is not None:
-                # display selected layer and return
-                self.DisplayLayer()
-                return
+            selected_layer = self.selected_layer[0]
+            selected_loc = self.selected_location[0]
+
+            if selected_layer is not None:
+                genv = gisenv()
+
+                # Check if the layer is in different location
+                if selected_loc.data['name'] != genv['LOCATION_NAME']:
+                    dlg = wx.MessageDialog(
+                        parent=self,
+                        message=_(
+                            "The map is not in the current location and cannot be displayed."
+                            "\n\n"
+                            "To view this map, switch to the selected mapset."
+                        ),
+                        caption=_("Unable to display the map"),
```

Comment:
Let's make the message little bit more specific:
"Map <mymap@mapset> is not in the current location and therefore cannot be displayed. \n\n To display this map switch to mapset <mapset> first."